include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+   * the flux
+   * @param id Identifier of the field with respect to which a derivative should
+   * be calculated
+   * @param jacobian Vector of the values of the derivative of the heat flux with
+   * respect to the field id
+   */
+  virtual void
+  vector_heat_flux_jacobian(
+    const std::map<field, std::vector<double>> &field_vectors,
+    const field                                 id,
+    std::vector<double>                        &jacobian_vector) = 0;
+
+  /**
+   * @brief momentum_flux Calculates the value of the evaporation recoil pressure.
+   * @param fields_value Value of the various field on which the recoil pressure
+   * may depend.
+   * @return value of the recoil pressure calculated with the fields_value
+   */
+  virtual double
+  momentum_flux(const std::map<field, double> &fields_value) = 0;
+
+  /**
+   * @brief vector_momentum_flux Calculates the values of the evaporation recoil
+   * pressure for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param momentum_flux_vector Vectors of the recoil pressure values
+   */
+  virtual void
+  vector_momentum_flux(
+    const std::map<field, std::vector<double>> &field_vectors,
+    std::vector<double>                        &momentum_flux_vector) = 0;
+
+  /**
+   * @brief momentum_flux_jacobian Calculates the jacobian (the partial derivative)
+   * of the evaporation recoil pressure with respect to a field
+   * @param field_values Value of the various fields on which the recoil pressure
+   * may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the recoil pressure with respect
+   * to the field.
+   */
+  virtual double
+  momentum_flux_jacobian(const std::map<field, double> &field_values,
+                         const field                    id) = 0;
+
+  /**
+   * @brief vector_momentum_flux_jacobian Calculate the derivative of the
+   * evaporation recoil pressure with respect to a field
+   * @param field_vectors Vector for the values of the fields on which the recoil
+   * pressure may depend.
+   * @param id Identifier of the field with respect to which a derivative should
+   * be calculated
+   * @param jacobian Vector of the value of the derivative of the recoil pressure
+   * with respect to the field id
+   */
+  virtual void
+  vector_momentum_flux_jacobian(
+    const std::map<field, std::vector<double>> &field_vectors,
+    const field                                 id,
+    std::vector<double>                        &jacobian_vector) = 0;
+
+protected:
+  // Map to indicate on which variables the model depends on
+  std::map<field, bool> model_depends_on;
+};
+
+/**
+ * @brief EvaporationModelConstant. Class that allows to calculate the
+ * evaporation heat and momentum fluxes when a constant mass flux is
+ * considered.
+ */
```

Comment:
Please consider adding @param for the parameters of the constructor :) I know we tend to forget that, but i think it's time we bcome better at this :)